### ISSUE TITLE
potential scripts processing issues in weechat_print modifier/modifier_data inputs

### ISSUE THREAD

**Issue by eau-u4f** [weechat/weechat#42]
Problem can appear with a channel name including ';' character in its name, it might introduce script parsing or unexpected processing result.
as said on irc, potential fix ideas:
@FlashCode with a pointer to buffer, instead of name

**Comment by flowbish** [comment]
Just to keep information all in one place.

This issue arose using the script colorize_nicks.py in a buffer with a semicolon in the name e.g. "ponychat.#asdfjkl;". Since the current buffer's name was incorrectly parsed, no nicks on the buffer were colorized.

**Comment by flashcode** [comment]
Not easy to fix this problem without backward incompatibility.

The solutions I can think of are:

**Solution 1:**

Escape the semicolons in the buffer name, for example: `irc;#wee\;chat;tags` (if the channel is `wee;chat`.

Drawbacks:
- breaks API (minor)
- scripts must be updated to properly split string, and it's hard to parse such string, the `\` must be taken into account.

**Solution 2:**

Add another modifier that sends "buffer_pointer;tags".

Drawbacks:
- scripts must be updated to use the new modifier
- weird to have 2 modifiers for the same thing
- introduces slowness in WeeChat (2 modifiers are sent per message displayed).

**Solution 3:**

As I already suggested, replace `plugin;buffer_name` by a buffer pointer (like `0x123abc`), for example: `0x123abc;tags`.
This definitely fixes the problem.

Drawbacks:
- breaks API (major)
- scripts must be updated to take care about the 2 possible formats, depending on the WeeChat version used.

I think the best is to implement **solution 3**, which fixes definitely the problem, but is not backward compatible.
So I'll do this and fix all scripts in repository to be compatible with both formats (only scripts that actually use the modifier data to find the buffer or use tags).
